### PR TITLE
[sailfish-browser] Use new 'Link:SetIcon' embedlite-components message. Fixes JB#50919

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

**Comment by rainemak** [comment]
Rebased on top of latest changes from "next". The commit message contains now a link the merge request as well.

File: src/qtmozembed/declarativewebpage.cpp [hunk 1]
```diff
@@ -65,7 +67,9 @@ DeclarativeWebPage::DeclarativeWebPage(QObject *parent)
     // subscribe to gecko messages
     std::vector<std::string> messages = { FULLSCREEN_MESSAGE,
                                           DOM_CONTENT_LOADED_MESSAGE,
-                                          LINK_ADDED_MESSAGE,
+                                          LINK_SET_ICON,
+                                          LINK_ADD_FEED,
+                                          LINK_ADD_SEARCH,
```
Comment (llewelld) [inline src/qtmozembed/declarativewebpage.cpp:72]: I don't see where "Link:AddFeed" or "Link:AddSearch" are used. Are they planned for future use?

File: src/pages/components/WebView.qml [hunk 2]
```diff
@@ -278,35 +276,12 @@ WebContainer {
...
-                            var faviconSize = parseInt(sizes[i])
-                            // Accept largest icon but one that is still smaller than Theme.itemSizeExtraLarge.
-                            if (faviconSize && faviconSize > iconSize && faviconSize <= Theme.itemSizeExtraLarge * 2) {
-                                iconSize = faviconSize
-                                parsedFavicon = true
-                            }
-                        }
-                    }
+                case "Link:SetIcon": {
+                  if (acceptedTouchIcon)
```
Comment (llewelld) [inline src/pages/components/WebView.qml:280]: Needs another couple of spaces indentation.

File: src/pages/components/WebView.qml [hunk 3]
```diff
@@ -278,35 +276,12 @@ WebContainer {
...
+                        return
 
-                    // Always pick at least one touch icon, parsed favicons are always based on size (good enough quality)
-                    if (!acceptedTouchIcon && acceptableTouchIcon || parsedFavicon) {
-                        favicon = data.href
-                        iconType = iconSize >= Theme.iconSizeMedium ? data.rel : ""
-                    }
+                    acceptedTouchIcon = data.isRichIcon
```
Comment (llewelld) [inline src/pages/components/WebView.qml:283]: Is the idea for using the `acceptedTouchIcon` flag that it forces use of the rich icon in preference to all others?

Comment (rainemak) [reply src/pages/components/WebView.qml:283]: Yes but there's necessarily no rich icon.

File: src/pages/components/WebView.qml (hunk 2 above)
Comment (rainemak) [reply src/pages/components/WebView.qml:280]: Fixed.

File: src/qtmozembed/declarativewebpage.cpp (hunk 1 above)
Comment (rainemak) [reply src/qtmozembed/declarativewebpage.cpp:72]: Yes. This is like it was previously.

File: src/pages/components/WebView.qml (hunk 2 above)
Comment (llewelld) [reply src/pages/components/WebView.qml]: Thanks

**Comment by llewelld** [comment]
When testing this it fails due to this: `JavaScript error: chrome://embedlite/content/ContentLinkHandler.jsm, line 343: NS_ERROR_UNEXPECTED: Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsIPrefBranch.getBoolPref]`

After commenting out the `getBoolPref` on [line 343](https://git.sailfishos.org/mer-core/embedlite-components/blob/f31d39f201233fc48ee8929fde60d09395f271f7/jsscripts/ContentLinkHandler.jsm#L343), the `Link:SetIcon` message now triggers as expected.

Is this anything to worry about?

**Comment by rainemak** [comment]
> When testing this it fails due to this: `JavaScript error: chrome://embedlite/content/ContentLinkHandler.jsm, line 343: NS_ERROR_UNEXPECTED: Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsIPrefBranch.getBoolPref]`
> 
> After commenting out the `getBoolPref` on [line 343](https://git.sailfishos.org/mer-core/embedlite-components/blob/f31d39f201233fc48ee8929fde60d09395f271f7/jsscripts/ContentLinkHandler.jsm#L343), the `Link:SetIcon` message now triggers as expected.
> 
> Is this anything to worry about?

Yes kind of. I had that defined :+1: 

Here is a PR:
https://git.sailfishos.org/mer-core/embedlite-components/merge_requests/80

File: src/qtmozembed/declarativewebpage.cpp (hunk 1 above)
Comment (llewelld) [reply src/qtmozembed/declarativewebpage.cpp:72]: Thanks for the explanation.

**Comment by llewelld** [comment]
Great, this now LGTM then.

**Pull request merged**